src/components/table/Table.tsx

```diff
@@ -1,23 +1,73 @@
-import { Table as AntdTable, TableProps } from 'antd'
+import { Table as AntdTable, SpinProps, TableProps } from 'antd'
 import cx from 'classnames'
 
 import s from './Table.module.css'
+import { Spinner } from '../spinner'
+import { CaretDown, CaretUp, CaretUpDown } from '@phosphor-icons/react'
 
 type MergedProps<T> = TableProps<T> & {
   verticalBorders?: boolean
+  columns: Exclude<TableProps<T>['columns'], undefined>
 }
 
 type Props<T> = Expand<MergedProps<T>>
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const Table = <T = any,>({ verticalBorders, ...props }: Props<T>) => {
+  const defaultLoadingProps: SpinProps = {
+    indicator: <Spinner className="!w-fit !h-fit !mx-0 translate-x-[-50%]" />,
+    tip: (
+      <div className="mt-20">
+        <h4>Loading this chart</h4>
+        <span className="text-lg">Please wait a moment</span>
+      </div>
+    ),
+  }
+
+  const loading: SpinProps | boolean | undefined =
+    typeof props.loading == 'object'
+      ? {
+          ...props.loading,
+          ...defaultLoadingProps,
+        }
+      : props.loading && defaultLoadingProps
+
+  const columns = props.columns.map((column) => ({
+    ...column,
+    sortIcon: ({ sortOrder }: { sortOrder: 'ascend' | 'descend' | null }) => {
+      if (!column.sorter) {
+        return
+      }
+
+      if (sortOrder == 'ascend') {
+        return <CaretUp />
+      } else if (sortOrder == 'descend') {
+        return <CaretDown />
+      }
+
+      return <CaretUpDown />
+    },
+  }))
+
   return (
     <div
       className={cx(props.className, {
         [s.bordered]: props.bordered && !verticalBorders,
       })}
     >
-      <AntdTable<T> {...props} />
+      <AntdTable<T>
+        showSorterTooltip={{ target: 'sorter-icon' }}
+        {...props}
+        loading={loading}
+        columns={columns}
+        locale={{
+          emptyText: (
```

Comment:
This represents that "No data" UI that is built-in to antd, but we just want all of the rows to be empty and just see the loading spinner

Reply:
Yeah! We get the spinner, but by default for some reason it has a "No data" UI behind the spinner too so this change removes that and just shows the table scaffolding and the loading spinner